reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,1588 @@
+            mask = dist <= np.median(dist) * 1.5
+            meta = meta.loc[mask]
+            meta['wind_gid'] = pos[mask]
+
+        return meta
+
+    @staticmethod
+    def _check_time_index(solar_time_index, wind_time_index):
+        """
+        Compare solar and wind time indexes. Determine coincident time_index
+        and requisite time_slices to reduce solar and wind dataset coincident
+
+        Parameters
+        ----------
+        solar_time_index : pandas.DatatimeIndex
+            Datetime index for solar data, respresents the temporal resolution
+            of the solar datasets
+        wind_time_index : pands.DatatimeINdex
+            Datetime index for wind data, respresents the temporal resolution
+            of the wind datasets
+
+        Returns
+        -------
+        time_index : pandas.DatatimeIndex
+            Coincident datetime index between solar and wind datasets,
+            i.e. datetime steps that are in both wind and solar data
+        solar_time_slice : slice | ndarray
+            slice or boolean index of the solar timesteps that are in the
+            coincident time_index
+        wind_time_slice : slice | ndarray
+            slice or boolean index of the wind timesteps that are in the
+            coincident time_index
+        """
+        solar_n = len(solar_time_index)
+        solar_time_slice = slice(None)
+        wind_n = len(wind_time_index)
+        wind_time_slice = slice(None)
+        if solar_n < wind_n:
+            wind_time_slice = wind_time_index.isin(solar_time_index)
+            time_index = solar_time_index
+            if not wind_time_slice.all():
+                msg = ("The following timesteps are not in both wind and "
+                       "solar dataset:\n{}"
+                       .format(wind_time_index[wind_time_slice]))
+                logger.error(msg)
+                raise RuntimeError(msg)
+        elif wind_n < solar_n:
+            solar_time_slice = solar_time_index.isin(wind_time_index)
+            time_index = wind_time_slice
+            if not solar_time_slice.all():
+                msg = ("The following timesteps are not in both wind and "
+                       "solar dataset:\n{}"
+                       .format(solar_time_index[solar_time_slice]))
+                logger.error(msg)
+                raise RuntimeError(msg)
+        else:
+            time_index = solar_time_index
+
+        return time_index, solar_time_slice, wind_time_slice
+
+    @staticmethod
+    def _groupby_data(data, diurnal=False, doy=False, month=False):
+        """
+        Groupby data by month and/or hour
+
+        Parameters
+        ----------
+        data : pandas.DataFrame
+            DataFrame of data where index is time_index, columns are sites
+        diurnal : bool, optional
+            Extract diurnal stats, by default False
+        doy : bool, optional
+            Extract doy-of-year stats, by default False
```

Comment:
If you're doing day-of-year statistics, should you add an option to roll the resource data to a local timezone? Otherwise the day of year is weird, right?